pandas/core/arrays/sparse/array.py

```diff
@@ -1636,6 +1639,44 @@ def _min_max(self, kind: Literal["min", "max"], skipna: bool) -> Scalar:
         else:
             return na_value_for_dtype(self.dtype.subtype, compat=False)
 
+    def _argmin_argmax(self, kind: Literal["argmin", "argmax"]) -> int:
+
+        values = self._sparse_values
+        mask = np.asarray(isna(self))[self._sparse_index.indices]
```

Comment:
Not necessarily for this PR, since this is already a big improvement, but does this `np.array(isna(self))` cause full array materialization? Could you avoid that by instead just computing `isna` on the sparse values themselves?

Reply:
This is an important point. Thanks.